### PR TITLE
[core] Skip duplicated IDs when querying point annotations

### PR DESCRIPTION
**Pull request by brunoabinader**
Extracted out of https://github.com/mapbox/mapbox-gl-native/pull/3563.

/cc @ivovandongen

### CODE REVIEW

File: src/mbgl/map/map.cpp [hunk 1]
```diff
@@ -755,7 +755,10 @@ AnnotationIDs Map::queryPointAnnotations(const ScreenBox& box) {
     for (auto &feature : features) {
         assert(feature.id);
         assert(*feature.id <= std::numeric_limits<AnnotationID>::max());
-        ids.push_back(static_cast<AnnotationID>(feature.id->get<uint64_t>()));
+        AnnotationID id = static_cast<AnnotationID>(feature.id->get<uint64_t>());
+        if (std::find(ids.begin(), ids.end(), id) == ids.end()) {
```
Comment (jfirebaugh) [inline src/mbgl/map/map.cpp]: This is an O(n²) algorithm. Better use a set to do deduplication.

**Comment by brunoabinader** [comment]
@1ec5 can you please review the relevant iOS and macOS parts?

File: src/mbgl/map/map.cpp (hunk 1 above)
Comment (brunoabinader) [inline src/mbgl/map/map.cpp]: > Now using AnnotationIDs as a typedef for std::set<AnnotationID>.

It turned out that platform code logic depended on particularities of `std::vector` e.g. ability to reassign member values via `std::remove_if` that would end up adding more code complexity by converting it to an approach in which `std::set` is compatible with.

Thus, I'm keeping `AnnotationIDs` as a `std::vector`. I'm using `std::set` internally when issuing the query and later moving those values to a `std::vector`, preserving linear complexity.

**Pull request merged**